edgedb authenticate

fantix opened the pull request.

tailhook commented on src/server/authenticate.rs:
```diff
@@ -0,0 +1,177 @@
+    }?;
+    if cred_path.exists() {
+        if cmd.assume_yes {
+            eprintln!("{} will be overwritten!", cred_path.display());
```
I would say `Owerwriting {filename}`. Since "will be" might be interpreted as "would be".

tailhook reviewed (commented):
Good start. But we also want:
1. Asking all credentials, that aren't specified in command-line (host, user, password)
2. Authorize the actual connection, to verify that credentials are good.

fantix commented:
Thanks for the review! I pushed a CRF and hopefully addressed most concerns.

> Authorize the actual connection, to verify that credentials are good.

`builder.connect_with_cert_verifier()` should've done it.

fantix commented on src/server/authenticate.rs:
```diff
@@ -0,0 +1,177 @@
+            }
+            Err(e) => {
+                // Bail out if we shouldn't ask the user to trust any cert
+                if !self.prompt {
```
I'm not sure I quite get the idea here - it's probably my bad to name it as `prompt`. In the CRF I renamed the variable to `system_ca_only` and added more comments.

tailhook replied:
I mean if `self.prompt` is false, why do use `InteractiveCertVerifier`, when you could have used `edgedb_client::tls::CertVerifier` and achieve the same result?

fantix replied:
Ah, that makes sense! Let me see here
